src/components/Post.jsx

```diff
@@ -550,13 +557,30 @@ function Post(prop) {
                                       {userComment.content.username}
                                     </strong>
                                     {userComment.content.text}
-                                    <span
-                                      onClick={(event) =>
-                                        deleteComment(event, userComment)
-                                      }
-                                    >
+                                    <span onClick={() => {
+                                      setOpenToDeleteComment(!openToDeleteComment)
+                                    }}>
+                                      {openToDeleteComment ?
+                                        <Dialog
+                                          fullScreen={fullScreen}
+                                          open={openToDeleteComment}
+                                          onClose={handleCloseForDeleteComment}
+                                          aria-labelledby="responsive-dialog-title">
+                                          <DialogTitle id="responsive-dialog-title">
+                                            {"Delete Comment?"}
+                                          </DialogTitle>
+                                          <DialogContent>
+                                            <DialogContentText>
+                                              Are you sure you want to delete this Comment?
+                                            </DialogContentText>
+                                          </DialogContent>
+                                          <DialogActions>
+                                            <Button onClick={handleCloseForDeleteComment}>Cancel</Button>
+                                            <Button onClick={(event) => deleteComment(event, userComment)}>Delete</Button>
```

Comment:
it's not deleting the selected comment,but oldest comment in that comment section. Some scope issue.

Reply:
fine, i will fix that